### PR TITLE
Add native impl of gelu for QuantizedCPU

### PR DESCRIPTION
**Pull request by Xia-Weiwen**
Add native implementation of gelu for quantized CPU.

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

### CODE REVIEW

**Comment by Xia-Weiwen** [comment]
The failure does not seem to be caused by this patch
```
20:30:48   test_fn_fwgrad_bwgrad_gradient_cuda_complex128 (__main__.TestGradientsCUDA) ... Memory exception on virtual address 0x7f586a00e000, node id 2 : Page not present
20:30:48 Address does not belong to a known buffer
20:30:48 Memory access fault by GPU node-2 (Agent handle: 0x561a5e5a6b50) on address 0x7f586a00e000. Reason: Page not present or supervisor privilege.
20:30:48 Traceback (most recent call last):
20:30:48   File "test/run_test.py", line 1068, in <module>
20:30:48     main()
20:30:48   File "test/run_test.py", line 1046, in main
20:30:48     raise RuntimeError(err_message)
20:30:48 RuntimeError: test_ops failed! Received signal: SIGIOT
```

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 1]
```diff
@@ -615,6 +615,55 @@ static void leaky_qrelu_out_kernel(Tensor& out, const Tensor& qx,
...
+              at::native::dequantize_val(scale, zero_point, value_qx);
+          const auto value_dy =
+              value_dx * 0.5 * (1 + std::erf(value_dx * M_SQRT1_2));
+          return at::native::quantize_val<scalar_t>(
+              output_scale, output_zero_point, value_dy);
+        },
+        [&](Vec value_qx) -> Vec {
+          const auto kAlphaVec = Vectorized<float>(M_SQRT1_2);
+          const auto kOneVec = Vectorized<float>(1);
+          const auto kPointFiveVec = Vectorized<float>(0.5);
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:654]: nit: maybe initialize outside the loop?

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 2]
```diff
@@ -615,6 +615,55 @@ static void leaky_qrelu_out_kernel(Tensor& out, const Tensor& qx,
...
+  // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
+  float scale = qx.q_scale();
+  auto scale_vec = Vectorized<float>(scale);
+  auto zero_point_vec = Vectorized<float>((float)zero_point);
+  auto scale_neg_zp_premul_vec = scale_vec * zero_point_vec.neg();
+  int64_t output_zero_point = zero_point;
+  float output_scale = scale;
+
+  AT_DISPATCH_QINT_TYPES(qx.scalar_type(), "qgelu", [&]() {
+    float inv_output_scale = 1.0 / output_scale;
```
Comment (vkuzo) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:629]: nit: init outside the macro?

File: test/quantization/core/test_quantized_op.py [hunk 3]
```diff
@@ -435,6 +435,31 @@ def test_qcelu(self, X, alpha):
...
+            dqX = qX.dequantize()
+
+            # torch.nn.functional
+            op = torch.nn.functional.gelu
+            dqY = op(dqX)
+            qY = torch.quantize_per_tensor(dqY, scale=scale, zero_point=zero_point,
+                                           dtype=torch_type)
+            qY_hat = op(qX)
+            self.assertEqual(qY.dequantize(), qY_hat.dequantize(),
+                             msg="F.leaky_relu failed ({} vs {})".format(qY, qY_hat))
```
Comment (vkuzo) [inline test/quantization/core/test_quantized_op.py:462]: nit: "gelu" instead of "leaky_relu"?

**Review by vkuzo — APPROVED** [review]
thanks, lgtm!

**Comment by vkuzo** [comment]
could you take a look at the minor comments, and I can import again after those?

**Comment by Xia-Weiwen** [comment]
> could you take a look at the minor comments, and I can import again after those?

Hi. I have modified the code according to your comments. Please take a look.

**Comment by vkuzo** [comment]
Hi @Xia-Weiwen , internal CI caught one more issue: "dos-newlines: You must use ONLY Unix linebreaks `\n` in source code." for qgelu.cpp.  Any chance you could check if the newline characters in that file are something other than `\n`, and change it to `\n`?

**Comment by Xia-Weiwen** [comment]
> Hi @Xia-Weiwen , internal CI caught one more issue: "dos-newlines: You must use ONLY Unix linebreaks `\n` in source code." for qgelu.cpp. Any chance you could check if the newline characters in that file are something other than `\n`, and change it to `\n`?

Hi @vkuzo. I have removed those '\r'. Please check again. Thanks.

**Comment by vkuzo** [comment]
any chance you could rebase and upload again?  I'm seeing a rebase conflict in Meta CI.

**Pull request closed without merging**